examples/repo_example/basic_repo.py

```diff
@@ -0,0 +1,397 @@
+# provides a generic 'Metadata' class. This class has two fields, one for
+# cryptographic signatures, i.e. 'signatures', and one for the payload over
+# which signatures are generated, i.e. 'signed'. The payload must be an
+# instance of either 'Targets', 'Snapshot', 'Timestamp' or 'Root' class. Common
+# fields in all of these 'Signed' classes are:
+#
+# spec_version -- The supported TUF specification version number.
+# version -- The metadata version number.
+# expires -- The metadata expiry date.
+#
+# The 'version', bumped on each metadata change, is used to reference metadata
+# from within other metadata, and thus allows for repository consistency in
+# addition to protecting against rollback attacks.
+#
+# The expiry date, protects against freeze attacks and allows for implicit key
+# revocation. Choosing an appropriate expiration interval depends on the
+# volatility of a role and how easy it is to re-sign them. Highly volatile
+# roles (timestamp, snapshot, delegated targets) usually have shorter
+# expiration intervals, whereas roles that change less and might use offline
+# keys (root, top-level targets) can expire less frequently.
+
+SPEC_VERSION = "1.0.19"
+
+# Define containers for role objects and cryptographic keys created below. This
+# allows us to sign and write metadata in a batch more easily.
+roles = {}
+keys = {}
+
+
+# Targets (integrity)
+# -------------------
+# The targets role guarantees integrity for the files that TUF aims to protect,
+# i.e. target files. It does so by listing the relevant target files, along
+# with their hash and length.
+roles["targets"] = Metadata[Targets](
+    signed=Targets(
+        version=1, spec_version=SPEC_VERSION, expires=_in(7), targets={}
+    ),
+    signatures=OrderedDict(),
+)
+
+# For the purpose of this example we use the top-level targets role to protect
+# the integrity of this very example script. The metadata entry contains the
+# hash and length of this file at the local path. In addition, it specifies the
+# 'target path', which a client uses to locate the target file relative to a
+# configured mirror base URL.
+#
+#      |----base URL---||-------target path-------|
+# e.g. tuf-examples.org/repo_example/basic_repo.py
+
+local_path = Path(__file__)
+target_path = f"{local_path.parts[-2]}/{local_path.parts[-1]}"
+
+target_file_info = TargetFile.from_file(target_path, local_path.resolve())
+roles["targets"].signed.targets[target_path] = target_file_info
+
+# Snapshot (consistency)
+# ----------------------
+# The snapshot role guarantees consistency of the entire repository. It does so
+# by listing all available targets metadata files at their latest version. This
+# becomes relevant, when there are multiple targets metadata files in a
+# repository and we want to protect the client against mix-and-match attacks.
+roles["snapshot"] = Metadata[Snapshot](
+    Snapshot(
+        version=1,
+        spec_version=SPEC_VERSION,
+        expires=_in(31),
```

Comment:
Observation: the comments earlier suggest that snapshot is usually a shorter expiration interval than targets, but the inverse is true for this code.

Reply:
Very good observation! My (admittedly not obvious) thinking here was that the top-level targets in the example is used *like* delegated targets, i.e. to list target files, and thus might change often, and thus might expire more quickly than a top-level target file that only delegates to other targets.

Do you think it is worth including this reasoning in the example, or does it go beyond its scope? I could try to update the comments and maybe also change the expiration interval below, when I make top-level targets a pure delegator.

Reply:
@joshuagl, do you think 83764c6 is enough?

Reply:
Seems more than sufficient, thank you!